src/shipit_static_analysis/shipit_static_analysis/batchreview.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import logging
+
+from rbtools.api.errors import APIError
+
+
+class BatchReview(object):
+    '''Create a review and comments with a single API call
+
+    Using BatchReview is much faster than creating a review and comments
+    with individual API calls.
+    '''
+
+    def __init__(self, api_root, review_request_id, diff_revision,
+                 max_comments=100, logger=None):
+        '''Initialize BatchReview
+
+        The ``api_root`` is the result of calling get_root on a Reviewboard
+        client.
+
+        The ``review_request_id`` is the integer identifier of the review
+        request on which to leave the review.
+
+        The ``diff_revision`` is the integer identifier of the diff
+        revision for which to leave the review.
+
+        The ``max_comments`` provides a limit on the number of comments
+        which can be made as part of the BatchReview.
+
+        The ``logger`` object is a logger to be used.
+        '''
+
+        self.api_root = api_root
+        self.review_request_id = review_request_id
+        self.diff_revision = diff_revision
+        self.max_comments = max_comments
+        self.comments = []
+
+        if logger is None:
```

Comment:
have logger be a global of this module, as done here -> https://github.com/mozilla-releng/services/blob/master/src/shipit_uplift/shipit_uplift/api.py#L27